src/main/java/org/jabref/logic/exporter/ModsExporter.java

```diff
@@ -76,267 +49,343 @@ public void export(final BibDatabaseContext databaseContext, final Path file, Li
                     } else if (StandardField.ISSN.equals(field)) {
-                        addIdentifier(StandardField.ISSN, value, mods);
+                        addIdentifier(writer, StandardField.ISSN, value);
                     } else if (StandardField.DOI.equals(field)) {
-                        addIdentifier(StandardField.DOI, value, mods);
+                        addIdentifier(writer, StandardField.DOI, value);
                     } else if (StandardField.PMID.equals(field)) {
-                        addIdentifier(StandardField.PMID, value, mods);
-                    } else if (StandardField.JOURNAL.equals(field)) {
-                        addJournal(value, relatedItem);
+                        addIdentifier(writer, StandardField.PMID, value);
+                    } else if (StandardField.PAGES.equals(field)) {
+                        addPart(parts, value);
+                    } else if (StandardField.VOLUME.equals(field)) {
+                        addPart(parts, value);
+                    } else if (StandardField.ISSUE.equals(field)) {
+                        addPart(parts, value);
                     }
-
-                    addOriginInformation(field, value, originInfo);
+                    trackOriginInformation(originItems, field, value);
                 }
-                mods.getModsGroup().add(originInfo);
 
-                addRelatedAndOriginInfoToModsGroup(relatedItem, partDefinition, mods);
-                modsCollection.getMods().add(mods);
-            }
+                writeOriginInformation(writer, originItems, fieldMap);
 
-            JAXBElement<ModsCollectionDefinition> jaxbElement = new JAXBElement<>(
-                    new QName(MODS_NAMESPACE_URI, "modsCollection"), ModsCollectionDefinition.class, modsCollection);
+                // Write related items
+                writeRelatedInformation(writer, parts, fieldMap);
+                writer.writeEndElement(); // end mods
+            }
 
-            createMarshallerAndWriteToFile(file, jaxbElement);
-        } catch (JAXBException ex) {
+            // end element and close
+            closeWriter(writer);
+        } catch (
+                XMLStreamException |
+                FileNotFoundException ex) {
             throw new SaveException(ex);
         }
     }
 
-    private void createMarshallerAndWriteToFile(Path file, JAXBElement<ModsCollectionDefinition> jaxbElement)
-            throws JAXBException {
+    private IndentingXMLStreamWriter createWriter(final Path file) throws XMLStreamException, FileNotFoundException {
+        XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();
+        FileOutputStream fOutputStream = new FileOutputStream(file.toFile());
+        IndentingXMLStreamWriter writer = new IndentingXMLStreamWriter(outputFactory.createXMLStreamWriter(fOutputStream));
+        writer.setIndentStep("    ");
+        writer.writeDTD("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n");
```

Comment:
i think, no \n necessary

![image](https://user-images.githubusercontent.com/1366654/235272641-0594c284-f655-4d15-bf9e-f6107f461ff2.png)

Reply:
IndentingStreamWriter is in com.sun.xml this might be problematic with jpackage. I used a transformer now